src/Nethermind/Nethermind.Blockchain/Producers/BaseLoopBlockProducer.cs

```diff
@@ -113,12 +116,24 @@ protected virtual async ValueTask ProducerLoopStep(CancellationToken cancellatio
         
         private void BlockTreeOnNewBestSuggestedBlock(object sender, BlockEventArgs e)
         {
-            CanProduce = false;
+            if (BlockTree.Head?.Hash != e.Block?.Hash)
+            {
+                Interlocked.Exchange(ref _canProduce, 0);
+                Metrics.CanProduceBlocks = 0;
```

Comment:
should you also do Interlocked.Exchange on metric? if you could have the other one corrupted then this one too and then metrics will confuse you